client-v2/src/redux/modules/pageViewData/actions.ts

```diff
@@ -112,28 +113,44 @@ const pageViewDataRequestedAction = (
   };
 };
 
-const fetchPageViewData = (
-  front: string,
-  articlesInFront: DerivedArticle[]
-): Promise<any> => {
-  return getPageViewDataFromOphan(buildRequestUrl(front, articlesInFront));
-};
-
-const buildRequestUrl = (
+const fetchPageViewData = async (
   frontId: string,
-  articles: DerivedArticle[]
-): string => {
-  const referringPath = `?referring-path=/${frontId}&`;
-  const articlePaths = articles
-    .map(article => `path=/${article.urlPath}&`)
-    .join('');
+  articlesInFront: string[]
+): Promise<PageViewDataFromOphan[]> => {
+  const base = 'https://fronts.local.dev-gutools.co.uk';
+  const referringPath = `/ophan/histogram?referring-path=/${frontId}&`;
   const timePeriod = `hours=${totalPeriodInHours}&interval=${intervalInMinutes}`;
-  return `${referringPath}${articlePaths}${timePeriod}`;
-};
+  const maxLength = 2048 - timePeriod.length - base.length;
+
+  const articlePaths = articlesInFront.map(article => `path=/${article}&`);
+
+  const urls: string[] = articlePaths.reduce(
+    (acc, path) => {
+      if (acc.length) {
+        const latestUrl = acc[acc.length - 1];
+        if (latestUrl.length + path.length < maxLength) {
```

Comment:
Minor: could we move `latestUrl` out of the `if` statement, and then do both comparisons in one go? If that works, it saves us a nested `if`, which is _slightly_ easier to wrap my head around 😁

Reply:
Yes, hopefully it makes more sense now! 🤞